src/test/java/org/opentripplanner/middleware/ApiKeyManagementTest.java

```diff
@@ -1,27 +1,126 @@
+        removeAdminUser(adminUser.id);
+
+        // refresh api keys
+        apiUser = getApiUser(apiUser.id);
+        for (ApiKey apiKey : apiUser.apiKeys) {
+            // delete api keys if present
+            String url = String.format("http://localhost:4567/api/secure/application/%s/apikey/%s",
+                apiUser.id,
+                apiKey.id);
+            getStatusCodeFromResponse(url, HttpUtils.REQUEST_METHOD.DELETE, apiUser.auth0UserId);
+        }
+
+        removeApiUser(apiUser.id);
+    }
+
+    /**
+     * Ensure that an {@link ApiUser} can create an API key for self.
+     */
+    @Test
+    public void canCreateApiKeyForSelf() {
+        String url = String.format("http://localhost:4567/api/secure/application/%s/apikey", apiUser.id);
+        Assertions.assertEquals(getStatusCodeFromResponse(url, HttpUtils.REQUEST_METHOD.POST, apiUser.auth0UserId), 200);
+    }
+
+    /**
+     * Ensure that an {@link AdminUser} can create an API key for an {@link ApiUser}.
+     */
+    @Test
+    public void adminCanCreateApiKeyForApiUser() {
+        String url = String.format("http://localhost:4567/api/secure/application/%s/apikey", apiUser.id);
+        Assertions.assertEquals(getStatusCodeFromResponse(url, HttpUtils.REQUEST_METHOD.POST, adminUser.auth0UserId), 200);
+    }
+
+    /**
+     * Ensure that an {@link ApiUser} can delete an API key for self.
+     */
+    @Test
+    public void canDeleteApiKeyForSelf() {
+
+        ensureAtLeastOneApiKeyIsAvailable();
+
+        // delete key
+        String url = String.format("http://localhost:4567/api/secure/application/%s/apikey/%s",
+            apiUser.id,
+            apiUser.apiKeys.get(0).id);
+        Assertions.assertEquals(getStatusCodeFromResponse(url, HttpUtils.REQUEST_METHOD.DELETE, apiUser.auth0UserId), 200);
     }
 
-    @Test @Disabled
-    public void getApiKeyForUser() {
+    /**
+     * Ensure that an {@link AdminUser} can delete an API key for an {@link ApiUser}.
+     */
+    @Test
+    public void adminCanDeleteApiKeyForApiUser() {
+
+        ensureAtLeastOneApiKeyIsAvailable();
+
+        // delete key
+        String url = String.format("http://localhost:4567/api/secure/application/%s/apikey/%s",
+            apiUser.id,
+            apiUser.apiKeys.get(0).id);
+        Assertions.assertEquals(getStatusCodeFromResponse(url, HttpUtils.REQUEST_METHOD.DELETE, adminUser.auth0UserId), 200);
     }
 
-    @Test @Disabled
-    public void deleteApiKeyForUser() {
+    /**
+     * Make sure that at least one API key exists.
+     */
+    private void ensureAtLeastOneApiKeyIsAvailable() {
+        // refresh api keys
+        apiUser = getApiUser(apiUser.id);
+
+        if (apiUser.apiKeys.isEmpty()) {
+            // create key
+            String url = String.format("http://localhost:4567/api/secure/application/%s/apikey", apiUser.id);
+            getStatusCodeFromResponse(url, HttpUtils.REQUEST_METHOD.POST, adminUser.auth0UserId);
```

Comment:
Since this is outside the context of a test (and is more so just setup activity), I don't think you need to do this via http. It might be worth refactoring a bit and adding an `ApiUser#createApiKey` method.

Reply:
I've updated to do this via ApiGatewayUtils.